src/acceptance/broker/broker_test.go

```diff
@@ -164,80 +167,84 @@ var _ = Describe("AutoScaler Service Broker", func() {
-		plans := getPlans()
-		if plans.length() < 2 {
-			Skip(fmt.Sprintf("2 plans needed, only one plan available plans:%+v", plans))
-			return
-		}
-		if !plans.contains("acceptance-standard") {
-			Skip(fmt.Sprintf("Acceptance test standard plan required plans:%+v", plans))
-			return
-		}
-
-		service := createService("acceptance-standard")
-		updateService := service.updatePlanRaw(plans[0])
-		Expect(updateService).To(Exit(1), "failed updating service")
+	Describe("allows updating service plans", func() {
+		var instance serviceInstance
+		It("should update a service instance from one plan to another plan", func() {
+			servicePlans := GetServicePlans(cfg)
+			source, target, err := servicePlans.getSourceAndTargetForPlanUpdate()
+			if err != nil {
+				Skip(err.Error())
+			}
+			instance = createService(source.Name)
+			instance.updatePlan(target.Name)
+		})
 
-		errStream := updateService.Err
-		if isCFVersion7() {
-			errStream = updateService.Out
-		}
-		Expect(string(errStream.Contents())).To(ContainSubstring("service does not support changing plans"))
-		service.delete()
+		AfterEach(func() {
+			instance.delete()
+		})
 	})
 })
 
-func isCFVersion7() bool {
-	version := cf.Cf("--version").Wait(cfg.DefaultTimeoutDuration())
-	Expect(version).To(Exit(0))
-	return strings.Contains(string(version.Out.Contents()), "cf version 7")
-}
-
-type plans []string
+type ServicePlans []ServicePlan
 
-func (p plans) length() int { return len(p) }
-func (p plans) contains(planName string) bool {
-	for _, plan := range p {
-		if plan == planName {
-			return true
-		}
+type (
+	ServicePlan struct {
+		Guid          string        `json:"guid"`
+		Name          string        `json:"name"`
+		BrokerCatalog BrokerCatalog `json:"broker_catalog"`
 	}
-	return false
-}
+	BrokerCatalog struct {
+		Id       string   `json:"id"`
+		Features Features `json:"features"`
+	}
+	Features struct {
+		PlanUpdateable bool `json:"plan_updateable"`
+	}
+)
+
+func (p ServicePlans) length() int { return len(p) }
 
-func getPlans() plans {
-	values := url2.Values{
-		"fields[service_offering.service_broker]": []string{"name"},
-		"include":                []string{"service_offering"},
+func GetServicePlans(cfg *config.Config) ServicePlans {
+	values := url.Values{
 		"per_page":               []string{"5000"},
 		"service_broker_names":   []string{cfg.ServiceBroker},
 		"service_offering_names": []string{cfg.ServiceName},
 	}
-	url := &url2.URL{Path: "/v3/service_plans", RawQuery: values.Encode()}
-	getPlans := cf.CfSilent("curl", url.String(), "-f").Wait(cfg.DefaultTimeoutDuration())
-	Expect(getPlans).To(Exit(0), "failed getting plans")
+	servicePlansURL := &url.URL{Path: "/v3/service_plans", RawQuery: values.Encode()}
+
+	var result ServicePlans
+
+	// This should also work as normal user - for some reason, if we use the normal user
+	// plan_updateable is returned as integer instead of boolean
```

Comment:
How did you find that out? 

I tried this with a regular user on another CloudFoundry landscape and got a boolean back for `plan_updateable`

Reply:
Yes, its pretty strange, and currently I don't have a minimal reproduction case that I could use to open a bug on the CF CLI for.

Reply:
Fine for me to do it like that since these are only tests 👼